credentialproviderpackage/pkg/templater/testdata/test_indent_template.yaml

```diff
@@ -0,0 +1,5 @@
+key1: {{ .Key1 }}
+key2: {{ .Key2 }}
+{{ if .Conditional }}
+{{ .KeyAndValue3 | indent 2 }}
+{{ end }}
```

Comment:
no newline at EOF

Reply:
This package and filewriter were copied from eks-anywhere. I would like to keep the consistency between the two

Reply:
Text files need to end with `\n` under the unix spec. If this comes from the partialyaml.go trimsuffix, we should remove that or fix the logic.